suite-native/firmware/src/components/UpdateProgressIndicator.tsx

```diff
@@ -0,0 +1,284 @@
+/* eslint-disable no-self-assign */
+import { useEffect, useMemo } from 'react';
+import { Platform } from 'react-native';
+import {
+    SharedValue,
+    useDerivedValue,
+    useSharedValue,
+    withDelay,
+    withSpring,
+    withTiming,
+} from 'react-native-reanimated';
+
+import {
+    Canvas,
+    Circle,
+    ColorMatrix,
+    Group,
+    ImageSVG,
+    MatrixColorFilterProps,
+    Paint,
+    Paragraph,
+    Path,
+    Shadow,
+    Skia,
+    TextAlign,
+    useSVG,
+} from '@shopify/react-native-skia';
+
+import { FirmwareOperationStatus } from '@suite-common/firmware';
+import { useNativeStyles } from '@trezor/styles';
+
+const CANVAS_SIZE = 160;
+const CIRCLE_RADIUS = 128;
+const CIRCLE_CENTER = CANVAS_SIZE / 2;
+
+const PROGRESS_STROKE_WIDTH = 5;
+
+const CHECKMARK_OFFSET_X = 10; // Adjust left/right position
+const CHECKMARK_OFFSET_Y = 0; // Adjust up/down position
+const CHECKMARK_SCALE = 1; // Adjust size
+const LONG_LEG_RATIO = 0.25; // Reduce from 0.33 (1/3) to make long leg shorter
+
+const checkmarkPath = Skia.Path.MakeFromSVGString(
+    `M${CIRCLE_CENTER - CIRCLE_RADIUS / 4 + CHECKMARK_OFFSET_X},${CIRCLE_CENTER + CHECKMARK_OFFSET_Y}` +
+        `l${(CIRCLE_RADIUS / 8) * CHECKMARK_SCALE},${(CIRCLE_RADIUS / 8) * CHECKMARK_SCALE}` +
+        `l${CIRCLE_RADIUS * LONG_LEG_RATIO * CHECKMARK_SCALE},-${CIRCLE_RADIUS * LONG_LEG_RATIO * CHECKMARK_SCALE}`,
+)!;
+
+const progressCirclePath = Skia.Path.MakeFromSVGString(
+    `M ${CIRCLE_CENTER},${CIRCLE_CENTER - (CIRCLE_RADIUS - PROGRESS_STROKE_WIDTH) / 2} A ${(CIRCLE_RADIUS - PROGRESS_STROKE_WIDTH) / 2},${(CIRCLE_RADIUS - PROGRESS_STROKE_WIDTH) / 2} 0 1,1 ${CIRCLE_CENTER},${CIRCLE_CENTER + (CIRCLE_RADIUS - PROGRESS_STROKE_WIDTH) / 2} A ${(CIRCLE_RADIUS - PROGRESS_STROKE_WIDTH) / 2},${(CIRCLE_RADIUS - PROGRESS_STROKE_WIDTH) / 2} 0 1,1 ${CIRCLE_CENTER},${CIRCLE_CENTER - (CIRCLE_RADIUS - PROGRESS_STROKE_WIDTH) / 2}`,
+)!;
+
+const fontStyle = {
+    fontFamily: Platform.select({ ios: 'Helvetica', default: 'serif' }),
```

Comment:
Good point, problem is that I need to load fonts as asset I am not sure if that woudn't cause that it will be included in bundle twice. Need to verify that.

Reply:
We are already using it in [DiscreetText](https://github.com/trezor/trezor-suite/blob/7e82d6e3f16502384c68bdc98fb99762d5ca59e2/suite-native/atoms/src/DiscreetText/DiscreetCanvas.tsx#L6-L7).

Reply:
Didn't know that!

Reply:
Percentages will be removed in next iteration so I will leave it as it is for now.